src/sage/categories/enumerated_sets.py

```diff
@@ -458,9 +458,12 @@ def __getitem__(self, i):
                 sage: F[1::2]
                 [2]
             """
+            from sage.rings.infinity import Infinity
             if isinstance(i, slice):
                 return self.unrank_range(i.start, i.stop, i.step)
             if i < 0:
+                i += self.cardinality()
+            if i is Infinity:
                 return self.list()[i]
```

Comment:
```suggestion
            if i is Infinity:
                raise ValueError("infinite list")
```
This will 100% fail with your current implementation, so let's short-circuit it as @user202729 is suggesting.